cpp/src/group_identity.cpp

```diff
@@ -0,0 +1,78 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <xtt-cpp/group_identity.hpp>
+
+#include "internal/text_to_binary.hpp"
+
+#include <xtt/crypto_wrapper.h>
+
+#include <stdexcept>
+
+using namespace xtt;
+
+group_identity::group_identity(const std::vector<unsigned char>& serialized)
+{
+    if (sizeof(xtt_group_id) != serialized.size()) {
+        throw std::runtime_error("Bad serialized value passed to group identity constructor");
```

Comment:
We should try to be exception free.  Generally, deserializers should be free functions (or static methods) that can return an error (or an `optional`), rather than constructors that have to throw.

This comment applies to all the wrapper classes.